### PR TITLE
Added CONTRIBUTING.md file

### PR DESCRIPTION
**Pull request by Ayush9026**
## Description 

This PR solved issue https://github.com/theupdateframework/theupdateframework.io/issues/50 . To enhance the contribution process and provide clear guidelines for new and existing contributors, we need to add a CONTRIBUTING.md file to the theupdateframework.io repository.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+1. **Request Assignment**: Comment on the issue asking for it to be assigned to you.
+2. **Prepare for Assignment**:
+    - Confirm that you have read the `CONTRIBUTING.md`.
+    - Ensure you have a functional development environment (you can build and run the project).
+    - Describe your intended approach to solving the issue.
+    - Include these points in one or more comments.
+3. **Start Work**: Begin working on the issue only after it has been assigned to you.
+4. **Avoid Conflicts**: Only start working on the issue and open a Pull Request after it has been assigned to you. This prevents confusion, duplicate work, and potential conflicts among contributors.
+5. **Reference the Issue**: In your Pull Request, reference the issue (e.g., `This PR fixes #1234`). This ensures the issue is automatically closed when your Pull Request is merged.
```
Comment (h4l0gen) [inline CONTRIBUTING.md:29]: Hi @Ayush9026, I think we don't have functionality of auto closing the issue. [See ](https://github.com/theupdateframework/theupdateframework.io/pull/39)even Referencing the issue in PR, we have to close it manually [see ](https://github.com/theupdateframework/theupdateframework.io/issues/37). So IMO we should remove this part. 

@lukpueh @joshuagl correct me if i am wrong 🙏

Comment (h4l0gen) [reply CONTRIBUTING.md:29]: @Ayush9026 you did a wonderful work, I've just a small suggestion, should we add testing guidelines too, on how to test changes locally same as [python-tuf CONTRIBUTING guideline.](https://theupdateframework.readthedocs.io/en/latest/CONTRIBUTING.html). Steps can be this:
Install Project Dependencies:

-  Execute the following command to install the project's dependencies locally:
   ` $ yarn install`
    
-  Start the Hugo development server to preview the website locally by running:
   `$ hugo server`
  
-   Once started, the website will be accessible at: http://localhost:1313/    by default.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+- Avoid repeating feedback, as it creates unnecessary noise. Check the existing conversation and use GitHub reactions to agree or disagree with a comment.
+- Do not approve pull requests without proper review just to improve your GitHub contributors graph.
+
+
+## Style Guide
+
+The theupdateframework.io site is built using HTML and Javascript.
+
+Check out the set of contributing guides available at https://theupdateframework.readthedocs.io/en/latest/CONTRIBUTING.html
```
Comment (h4l0gen) [inline CONTRIBUTING.md:61]: Also, why referencing `python-tuf`'s [Instructions for contributors](https://theupdateframework.readthedocs.io/en/latest/CONTRIBUTING.html) here? That page contains 

- DCO
- Testing 
- Unit tests
- Coverage
- Auto formatting

Which completely based on `python-tuf` repo, does not relates to `theupdateframework.io` repository. 

Thank you 👍

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
+# Contributing to theupdateframework.io
```
Comment (h4l0gen) [inline CONTRIBUTING.md:1]: @Ayush9026 you did a wonderful work, I've just a small suggestion, should we add testing guidelines too, on how to test changes locally same as [python-tuf CONTRIBUTING guideline.](https://theupdateframework.readthedocs.io/en/latest/CONTRIBUTING.html). Steps can be this:


-  Execute the following command to install the project's dependencies locally:
   ` $ yarn install`
    
-  Start the Hugo development server to preview the website locally by running:
   `$ hugo server`
  
-   Once started, the website will be accessible at: http://localhost:1313/    by default. And review your changes.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+
+> **Notes:**
+>
+> - **Check the Assignees Box**: Before requesting to be assigned to an issue, check the `Assignees` box at the top of the page to see if it has already been assigned to someone else. If it has a current assignee but appears to be inactive, politely ask them if they are still working on it or if you might collaborate with them.
+> - **Requesting Assignment**: Only request to be assigned to an issue if you know how to work on it.
+> - **Seek Clarity**: If an issue is unclear, ask questions to get more information before requesting to be assigned. Avoid asking vague questions like "what do I do next?" or "how do I fix this?"
+> - **Collaboration**: An issue can be assigned to multiple people if all parties agree to collaborate on it. The pull request can contain commits from different collaborators.
+> - **Inactivity**: Any issue with no activity for 2 weeks will be unassigned and re-assigned to someone else.
+
```
Comment (h4l0gen) [inline CONTRIBUTING.md:39]: IMO we should add `DCO` section clearly in `CONTRIBUTING.md`. In every CNCF project it is included see [here ](https://theupdateframework.readthedocs.io/en/latest/CONTRIBUTING.html#dco)in `python-tuf`, just for [reference](https://github.com/cncf/foundation/blob/main/dco-guidelines.md#dco-guidelines-v10). WYT? @Ayush9026 
Any suggestions @lukpueh @joshuagl

File: CONTRIBUTING.md (hunk 1 above)
Comment (Ayush9026) [reply CONTRIBUTING.md:29]: Hi @h4l0gen,

Thanks 🙏 for letting me know. If auto-closing isn't supported, I'll remove that part once @lukpueh and @joshuagl review it.

File: CONTRIBUTING.md (hunk 2 above)
Comment (Ayush9026) [reply CONTRIBUTING.md:61]: Hi @h4l0gen,

Thank you for the feedback! 👍 You’re right, referencing the python-tuf’s instructions isn't relevant here. I’ll update the contributing guide to better reflect the specifics of the theupdateframework.io repository. 

Thanks for pointing that out! 🙌

Comment (h4l0gen) [reply CONTRIBUTING.md:61]: Please just wait for official maintainer's review, before making any commit. I am not official maintainer, so there review is must required on if my review is correct or not. I hope you understand. Thank you 👍

File: CONTRIBUTING.md (hunk 3 above)
Comment (Ayush9026) [reply CONTRIBUTING.md:1]: Hi @h4l0gen,

Thank you for the kind words! 😊 That's a great suggestion. Adding testing guidelines for local changes will definitely help contributors. I'll include the steps you mentioned once @lukpueh and @joshuagl review it.

Thanks again for your valuable feedback! 🙌

File: CONTRIBUTING.md (hunk 4 above)
Comment (Ayush9026) [reply CONTRIBUTING.md:39]: That's a great point. Including a DCO section would align with CNCF project standards and provide clarity for contributors. I’ll add this section to the CONTRIBUTING.md once @lukpueh and @joshuagl sir have reviewed it.

Thank you for the suggestion! 🙏

File: CONTRIBUTING.md (hunk 2 above)
Comment (Ayush9026) [reply CONTRIBUTING.md:61]: Hi @h4l0gen ,

Of course, I understand. I'll wait for the official maintainers' review before making any changes.

**Comment by Ayush9026** [comment]
@lukpueh and @joshuagl sir please review it.

**Pull request closed without merging**